streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java

```diff
@@ -521,6 +521,10 @@ public synchronized StreamsBuilder addStateStore(final StoreBuilder<?> builder)
      * <p>
      * It is not required to connect a global store to {@link Processor Processors}, {@link Transformer Transformers},
      * or {@link ValueTransformer ValueTransformer}; those have read-only access to all global stores by default.
+     * <p>
+     * The supplier should always generate a new instance each time invoking {@link  ProcessorSupplier#get()}. Creating
```

Comment:
nit `each time {@link  ProcessorSupplier#get()} is called.` (similar elsewhere -- please fix throughout the whole PR)